src/util/secrets/secrets.h

```diff
@@ -95,15 +95,18 @@ errno_t sss_sec_list(TALLOC_CTX *mem_ctx,
 
 errno_t sss_sec_get(TALLOC_CTX *mem_ctx,
                     struct sss_sec_req *req,
-                    char **_secret);
+                    char **_secret,
+                    char **_datatype);
 
 errno_t sss_sec_put(struct sss_sec_req *req,
                     const char *secret,
-                    enum sss_sec_enctype enctype);
+                    enum sss_sec_enctype enctype,
+                    const char *datatype);
```

Comment:
I think it would be nice to have some explanation in the commit message: why is this needed to be able to specify datatype.

Reply:
And the same about switch to binary format.

Reply:
Done.